### ISSUE TITLE
Provide a migration path towards native promises

### ISSUE THREAD

**Issue by spion** [petkaantonov/bluebird#1593]
With node 12 adding support for async stack traces, I think it has become sensible for some bluebird users to migrate towards native promises. It would be nice to them if we provided a less painful migration path, and to describe which features allow for migration and which don't.

I'll try and start enumerating the ones I don't think are possible to implement as static methods for native promises:

1. Cancellation
2. Synchronous inspection (although you could "lie": use a WeakMap and the first time synchronous inspection is attempted return "pending" add a handler to the promise)
3. Monitoring (although probably possible in node using async_hooks: https://nodejs.org/api/async_hooks.html#async_hooks_promise_execution_tracking)

**Comment by benjamingr** [comment]
I started this with https://www.npmjs.com/package/bluebird-api but never really spent a lot of time on it - let me know if you want to pick it up.

**Comment by benjamingr** [comment]
Also, it's also fine to never migrate and just do:

```js
Bluebird.setScheduler(fn => Promise.resolve().then(fn))
```

To use the browser's native scheduling. Async functions use native promises anyway.

Also note that async stack traces only work with async functions so bluebird actually still provides better async stack traces for promise chains.

**Comment by spion** [comment]
Its not really fine. Keeping versions of bluebird in sync across modules - especially when developing using npm link - is still a huge pain. Async functions always returning regular promises which have none of the extra methods is also a pain. None of this is Bluebird's fault, but we have to work with what we've got.

**Comment by benjamingr** [comment]
@spion right, but bluebird static methods work just fine in regular non-bluebird code and interop is seamless. If native promises had async stack traces in production I agree but we only have those for async functions.

I'd argue most libraries are still better off with bluebird than with native promises.

**Comment by benjamingr** [comment]
Also, given the number of major releases of bluebird we do in a year (0) and the number of patches (like 3-4 that are all compatible) and how stable things are I find "hard to keep in sync" a hard sell.

**Comment by spion** [comment]
Its not that, there is e.g. poor TypeScript behavior when you're developing a `npm link`-ed module that depends on bluebird and a project that also depends on it.

Again, its also not that simple to just mix and match bluebid and native promises. Monitoring, async hooks, stack traces etc will all suffer from it.
